c2corg_api/tests/views/test_articles.py

```diff
@@ -0,0 +1,435 @@
+
+    def test_get_lang(self):
+        self.get_lang(self.article1)
+
+    def test_get_new_lang(self):
+        self.get_new_lang(self.article1)
+
+    def test_get_404(self):
+        self.get_404()
+
+    def test_post_error(self):
+        body = self.post_error({}, user='moderator')
+        errors = body.get('errors')
+        self.assertEqual(len(errors), 2)
+        self.assertCorniceRequired(errors[0], 'locales')
+        # self.assertCorniceRequired(errors[1], 'geometry')
+
+    def test_post_missing_title(self):
+        body_post = {
+            'categories': ['site_info'],
+            'activities': ['hiking'],
+            'article_type': 'collab',
+            'locales': [
+                {'lang': 'en'}
+            ]
+        }
+        body = self.post_missing_title(body_post, user='moderator')
+        errors = body.get('errors')
+        self.assertEqual(len(errors), 2)
+        self.assertCorniceRequired(errors[0], 'locales.0.title')
+        self.assertCorniceRequired(errors[1], 'locales')
+
+    def test_post_non_whitelisted_attribute(self):
+        body = {
+            'article_type': 'collab',
+            'protected': True,
+            'locales': [
+                {'lang': 'en', 'title': 'Lac d\'Annecy'}
+            ]
+        }
+        self.post_non_whitelisted_attribute(body, user='moderator')
+
+    def test_post_missing_content_type(self):
+        self.post_missing_content_type({})
+
+    def test_post_success(self):
+        body = {
+            'document_id': 123456,
+            'version': 567890,
+            'categories': ['site_info'],
+            'activities': ['hiking'],
+            'article_type': 'collab',
+            'associations': {
+                'waypoints': [
+                    {'document_id': self.waypoint2.document_id}
+                ],
+                'articles': [
+                    {'document_id': self.article2.document_id}
+                ]
+            },
+            'geometry': {
+                'version': 1,
+                'document_id': self.waypoint2.document_id,
+                'geom':
+                    '{"type": "Point", "coordinates": [635956, 5723604]}'
+            },
+            'locales': [
+                {'lang': 'en', 'title': 'Lac d\'Annecy'}
+            ]
+        }
+        body, doc = self.post_success(body, user='moderator')
+        version = doc.versions[0]
+
+        archive_article = version.document_archive
+        self.assertEqual(archive_article.categories, ['site_info'])
+        self.assertEqual(archive_article.activities, ['hiking'])
+        self.assertEqual(archive_article.article_type, 'collab')
+
+        archive_locale = version.document_locales_archive
+        self.assertEqual(archive_locale.lang, 'en')
+        self.assertEqual(archive_locale.title, 'Lac d\'Annecy')
+
+        # check if geometry is not stored in database afterwards
+        query_article = self.session.query(Article).get(
```

Comment:
Yes, that is correct. It's only that `doc` is the same as `query_article`. So you can directly test with `doc.geometry`.